lib/wakes.rb

```diff
@@ -39,8 +39,12 @@ def self.create!(*args)
     build(*args).tap(&:save!)
   end
 
-  def self.redirect(source, target, label = nil)
-    RedirectMapper.new(source, target, label)
+  def self.redirect(source_path, target_path, label = nil)
+    RedirectMapper.new(source_path, target_path, label)
+  end
+
+  def self.redirect_with_hosts(source_host: nil, source_path:, target_host: nil, target_path:, label: nil)
```

Comment:
gotcha. So from an API standpoint, we don't care what Ruby's `URI` thinks about it :). We can put it in the API... it'll be more useful than the one that includes the protocol. We're going to strip out the protocol, anyways.